rosbag2_transport/include/rosbag2_transport/player.hpp

```diff
@@ -160,6 +194,14 @@ class Player : public rclcpp::Node
   rclcpp::Service<rosbag2_interfaces::srv::GetRate>::SharedPtr srv_get_rate_;
   rclcpp::Service<rosbag2_interfaces::srv::SetRate>::SharedPtr srv_set_rate_;
   rclcpp::Service<rosbag2_interfaces::srv::PlayNext>::SharedPtr srv_play_next_;
+
+  // defaults
+  std::shared_ptr<KeyboardHandlerBase> keyboard_handler_;
+  std::vector<KeyboardHandlerBase::callback_handle_t> keyboard_callbacks_;
+
+  int num_paused_ = 0;
```

Comment:
these counts seem purely for testing purposes (unless i'm misreading). You should be able to expect call counts via `EXPECT_CALL().Times()` for the Mock KeboardHandler methods without adding extra instrumentation to the production code. See https://google.github.io/googletest/gmock_cheat_sheet.html#UsingMocks

Reply:
so i'm not trying to mock the player, but that the callbacks perform correctly for the real player. i just pulled it into a wrapper class.